data_structures/bloom_filters/bloom_filter/bloom_filter.c

```diff
@@ -0,0 +1,146 @@
+#include "./bloom_filter.h"
+
+#include <math.h>
+
+static inline int murmur_32_scramble(int k)
+{
+    k *= 0xcc9e2d51;
+    k = (k << 15) | (k >> 17);
+    k *= 0x1b873593;
+    return k;
+}
+int murmur3_32(const uint8_t *key, size_t len, int seed)
```

Comment:
It's pretty clever to simulate multiple hash functions with one hash function and multiple seeds, but maybe you should note that explicitly in a comment, can be somewhat confusing otherwise